sphinx/ext/autodoc/importer.py

```diff
@@ -156,6 +156,17 @@ def import_object(modname, objpath, objtype='', attrgetter=safe_getattr, warning
 
     try:
         module = import_module(modname, warningiserror=warningiserror)
+        import os
+        if hasattr(module, '__file__') and os.path.isfile('{}i'.format(module.__file__)):
+            try:
+                from importlib._bootstrap import spec_from_loader
+                from importlib._bootstrap_external import SourceFileLoader
+                from importlib._bootstrap import module_from_spec
```

Comment:
Note: these functions are provided by `importlib.util` package. I feel `importlib._bootstrap` is private and non-public interface. So this should be refactored later.
https://docs.python.org/3.6/library/importlib.html#importlib.util.spec_from_loader